contracts/message-based/PriceAwareUpgradeable.sol

```diff
@@ -107,32 +119,42 @@ contract PriceAwareUpgradeable is OwnableUpgradeable {
 
         require(block.timestamp - dataTimestamp < maxDelay, "Data is too old");
 
+        //We iterate directly through call data to extract the values of symbols
 
-        //Debugging logs (to be removed)
-
-        //    console.log("Len: ", messageLength);
-        //    console.logBytes(rawData);
-        //    console.logBytes32(hash);
-        //    console.logBytes(signature);
-        //    console.log("Signer: ", signer);
-
+        return _readFromCallData(symbols, uint256(dataSize), messageLength);
+    }
 
-        //8. We iterate directly through call data to extract the value for a given symbol
 
-        uint256 val;
-        uint256 max = dataSize;
-        bytes32 currentSymbol;
+    function _readFromCallData(bytes32[] memory symbols, uint256 dataSize, uint16 messageLength) private view returns (uint32[] memory) {
+        bytes32[] memory resultSymbols;
```

Comment:
Do wy use it?

Reply:
Do you mean this method? It's because before would throw `Stack to deep` error. Probably could be solved in other way as well

Reply:
not anymore